connector_salesforce/README.rst

```diff
@@ -0,0 +1,247 @@
+ - Import of Salesforce priceBook entries into Odoo pricelist version. Salesforce is the master.
+ - Import of won opportunities into Odoo. Salesforce is the master.
+
+Installation
+============
+
+To install this module, you need to:
+
+ * install simplesalesforce
+
+    `pip install simple-salesforce`
+
+ * install requests
+
+    `pip install requests`
+
+
+Configuration
+=============
+
+Configure Connection With Salesforce
+------------------------------------
+
+There are 2 supported ways to access a Salesforce instance with the Salesforce connector
+
+Oauth2 flow
+***********
+
+In Salesforce
+#############
+
+In order to use it you have to add a remote application in Salesforce administration
+and enable Oauth2 login. The app should preferably be named `odoo_connector`
+
+The created remote access app must have following parameters:
+
+  Permitted Users --> All users may self-authorize
+  Callback URL --> public_odoo_url/salesforce/oauth
+
+
+Once done, you have to manage your app and ensure the
+`Refresh token is valid until revoked` parameter is set
+and configure the desired authorization of the app.
+
+In Odoo
+#######
+
+Go to the menu `Connector -> Salesforce -> Salesforce backend` and create a new backend.
+Go to the Authentication Pane of your backend and enter the following parameters:
+
+ - `Authentication Method`: Choose oauth2
+ - `URL`: Enter the URL of your Salesforce instance
+ - `Connect on sandbox instance`: Check it if you are working on a sandox Salesforce instance
+ - `OAuth2 Consumer Key`: Enter the consumer key provided by Salesforce app
+ - `OAuth2 secret`: Enter the consumer secret provided by Salesforce app
+ - `Public secure URL of Odoo (HTTPS)`: Enter the callback URL you have enterd by Salesforce
+ - `OAuth2 client authorization code`: Enter the authorization code provided by Salesforce app
+
+Choose a default shop in `Import/Export Configuration` pane and save your backend.
+
+Then simply press `the Authorize Odoo in Salesforce` and follow the instruction provided
+in Salesforce. Once done, your application must be authorized in Salesforce and tokens
+must be present in backend.
+
+!!!Only the OAuth flow is production proof. Other Authentication methods are still exprimental!!!
+
+
+User Password flow
+******************
+
+This flow allows a user to connect to api using SOAP access
+in order to get a token. This approach is simpler but less secure.
+The first step is to pass the domain of your Salesforce instance
+and an access token straight in Odoo backend.
+
+You must have the full URL e.g (https://na1.salesforce.com) of your instance.
+
+There are also two means of authentication:
+ - Using username, password and security token
+ - Using IP filtering, username, password and organizationId
+
+Simply choose corresponding authentication method and
+set required fields.
+
```

Comment:
Security Warning: This approach is simpler but less secure than Oauth authentication process.